src/routing/approved.rs

```diff
@@ -836,14 +837,21 @@ impl Approved {
         }
 
         debug!(
-            "Received Relocate message to join the section at {}.",
+            "Received Relocate message to join the section at {}",
             details.relocate_details().destination
         );
 
-        if self.relocate_state.is_none() {
-            self.send_event(Event::RelocationStarted {
-                previous_name: self.node.name(),
-            });
+        match self.relocate_state {
+            Some(RelocateState::InProgress(_)) => {
```

Comment:
The commit message might be imprecise. The idea is we want to avoid multiple concurrent relocations of the same node (so a node is relocated to a destination X and while that is being processed, it gets relocated again to a different destination Y). But we still allow multiple sequential relocations (that is, node is first relocated to section A, then some time later relocated again to section B, and so on...)